flink-end-to-end-perf-tests/flink-basic-operations/src/main/java/org/apache/flink/basic/operations/PerformanceTestJob.java

```diff
@@ -0,0 +1,225 @@
+		} else if ("broadcast".equals(streamPartitioner)) {
+			outputNode = flapNode.broadcast();
+		} else {
+			throw new IllegalArgumentException("Argument streamPartitioner is illegal!");
+		}
+		return outputNode;
+
+	}
+
+	protected DataStream<Tuple3<String, String, String>> connector(
+		DataStream<Tuple2<String, String>> output1,
+		DataStream<Tuple2<String, String>> output2,
+		int parallelism){
+		DataStream<Tuple3<String, String, String>> output = output1.connect(output2).process(
+			new CoProcessFunction<Tuple2<String, String>, Tuple2<String, String>, Tuple3<String, String, String>>() {
+				@Override
+				public void processElement1(Tuple2<String, String> value, Context ctx, Collector<Tuple3<String, String, String>> out) throws Exception {
+					out.collect(new Tuple3<>(value.f0, value.f1, value.f1));
+				}
+
+				@Override
+				public void processElement2(Tuple2<String, String> value, Context ctx, Collector<Tuple3<String, String, String>>out) throws Exception {
+					processElement1(value, ctx, out);
+				}
+
+			}, new TupleTypeInfo<>(BasicTypeInfo.STRING_TYPE_INFO, BasicTypeInfo.STRING_TYPE_INFO,
+				BasicTypeInfo.STRING_TYPE_INFO)).name("downStreamWithInputOrder").setParallelism(parallelism);
+		return output;
+	}
+
+	public static void main(String[] args) throws Exception {
+
+		ParameterTool paramTools = ParameterTool.fromArgs(args);
+		String jobName = paramTools.get("jobName", "Basic PerformanceTestJob");
+		int parallelism = paramTools.getInt("parallelism", 1);
+		String outputPath = paramTools.get("outputPath", "");
+		String checkPointMode = paramTools.get("checkPointMode", "AtLeastOnce").toLowerCase();
+		int checkPointTimeout = paramTools.getInt("checkPointTimeout", 60 * 15);
+		long maxCount = paramTools.getLong("maxCount", Long.MAX_VALUE);
+		int seed = paramTools.getInt("seed", 0);
+		int sleepNum = paramTools.getInt("sleepNum", 0);
+		String streamPartitioner = paramTools.get("streamPartitioner", "KeyBy").toLowerCase();
+		String scheduleMode = paramTools.get("scheduleMode", "EAGER").toLowerCase();
+		String executionMode = paramTools.get("executionMode", "PIPELINED").toLowerCase();
+		long checkpointInterval = paramTools.getLong("checkpointInterval", 0L);
+		String checkpointPath = paramTools.get("checkpointPath", "/");
+		String stateBackend = paramTools.get("stateBackend", "heap").toLowerCase();
+		long checkpointTimeout = paramTools.getLong("checkpointTimeout", 15 * 1000L);
+		int recordSize = paramTools.getInt("recordSize", 1024);
+		String topologyName = paramTools.get("topologyName", "oneInput").toLowerCase();
+
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.getCheckpointConfig().setCheckpointTimeout(checkPointTimeout);
+		env.setStreamTimeCharacteristic(TimeCharacteristic.IngestionTime);
+		env.setParallelism(parallelism);
+		env.setMaxParallelism(30720);
+
+		PerformanceTestJob testJob = new PerformanceTestJob();
+		if ("oneinput".equals(topologyName)){
```

Comment:
I'm really questioning the design here; I have a hard time understanding where the line is between customization via parameters or overrides.

Reply:
I'm sorry for costing you much time to understand it. If you participated in the discussion of the previous plan, it may be easier for you to understand.
In the two topological scenarios, there are too many overlapping parts, so for the convenience of writing the job, different topological names are distinguished by a parameter.